client/allocrunner/consul_hook.go

```diff
@@ -98,13 +98,12 @@ func (h *consulHook) Prerun() error {
 }
 
 func (h *consulHook) prepareConsulTokensForTask(job *structs.Job, task *structs.Task, tgName string, tokens map[string]map[string]string) error {
-	var consulClusterName string
-	if task.Consul != nil && task.Consul.Cluster != "" {
-		consulClusterName = task.Consul.Cluster
-	} else {
-		consulClusterName = structs.ConsulDefaultCluster
+	if task.Consul == nil {
+		return nil
 	}
```

Comment:
`task.Consul` can be nil if the task group's `Consul` block is non-nil, so I think we need to check there too.

Reply:
but do we ever use task group's `Consul` block to get the Consul cluster name? I don't think we do?

Reply:
Ah, we're probably not at the moment. I'll make sure that lands with the rest of the multi-cluster client work.

Reply:
just to be clear, what I mean is that this method, `prepareConsulTokensForTask` is only called for task group tasks, and the only reason we're doing a `nil` check here is because we need the Consul cluster name. I'm just trying to think of  a situation in which `task.Consul` would be `nil` and we'd still need to get a Consul token for it.